enterprise/server/remote_execution/persistentworker/persistentworker.go

```diff
@@ -79,8 +80,19 @@ func Start(ctx context.Context, workspace *workspace.Workspace, container contai
 		stdinWriter:  stdinWriter,
 		stdoutReader: bufio.NewReader(stdoutReader),
 	}
+
+	// Use the same size limit as regular actions to prevent OOM from persistent worker stderr
+	if *commandutil.StdOutErrMaxSize > 0 {
+		w.stderr = *lockingbuffer.NewWithMaxSize(*commandutil.StdOutErrMaxSize)
+	} else {
+		w.stderr = *lockingbuffer.New()
+	}
 	if protocol == jsonProtocol {
-		w.jsonDecoder = json.NewDecoder(stdoutReader)
+		if *commandutil.StdOutErrMaxSize > 0 {
+			w.jsonDecoder = json.NewDecoder(io.LimitReader(w.stdoutReader, int64(*commandutil.StdOutErrMaxSize)))
```

Comment:
`LimitReader` returns EOF when it reaches the limit, rather than an Error (e.g. ResourceExhausted). So I believe this would result in confusing JSON parse errors below, when the JSON output unexpectedly ends. Wonder if we should provide a Reader impl in the commandutil package similar to the limitWriter impl